server/src/main/java/org/elasticsearch/index/IndexVersions.java

```diff
@@ -128,9 +128,8 @@ private static Version parseUnchecked(String version) {
     public static final IndexVersion MERGE_ON_RECOVERY_VERSION = def(8_515_00_0, Version.LUCENE_9_11_1);
     public static final IndexVersion UPGRADE_TO_LUCENE_9_12 = def(8_516_00_0, Version.LUCENE_9_12_0);
     public static final IndexVersion ENABLE_IGNORE_ABOVE_LOGSDB = def(8_517_00_0, Version.LUCENE_9_12_0);
-
     public static final IndexVersion UPGRADE_TO_LUCENE_10_0_0 = def(9_000_00_0, Version.LUCENE_10_0_0);
-
+    public static final IndexVersion LOGSDB_DEFAULT_IGNORE_DYNAMIC_BEYOND_LIMIT = def(9_000_00_1, Version.LUCENE_10_0_0);
```

Comment:
The way I see it is

* No issue for users who have explicitly set the value (whether `false` or `true`) since that overrides the default.

* Change in behavior for users relying on the default false: After upgrading, documents that would have been 
rejected (due to exceeding the field limit) will now be indexed with extra fields ignored.

While this change doesn't technically "break" anything, it may surprise users who were relying on documents being rejected in case of mapping issues. Moving from default `false` to default `true` is eventually going to accept more documents which would have been rejected before.

I am fine backporting only to `8.17` anyway.

Reply:
I understand your reasoning and it makes sense. I suspect that just technically this may be hard to backport even to 8.17. F.e. `UPGRADE_TO_LUCENE_10_0_0` does not exist in 8.x. We may need guidance from someone on this.

Reply:
Based on recent version in this file, shouldn't new version be: `9_001_00_0`? Since we're are not creating a patch version? 

When back porting, I don't think the same version id can be used. Based on the latest index version in 8.x, I believe this should be: `8_518_00_0 `. However after reading the comment in `IndexVersions` class, I don't fully understand how the main branch should updated after the backport has merged. I think a new index version constant needs to be added with version id `8_518_00_0 `? We have then two versions, for stateful and stateless. But I may be wrong here. @thecoop  would you be able to give advice here what index versioning and back porting now that main uses a different major index version?

Additionally I also think we don't need to backport to this 8.16 branch.

Reply:
I would expect that when backporting this we have a compiler failure because the index version I added here does not exist in the other branch. In that case I expect for the backport I just need to add the appropriate new version (the next one). SOmething like `8_518_00_0` or a newer one if this one is already created by another backport.

Reply:
When backporting, you need to introduce a new version constant `8_518_00_0` which represents 'this change, but with lucene 9' as a backport, and checks on the 8.x branch need to use this constant like `indexVersion >= BACKPORT_CONSTANT`

You then need to also understand that new constant on `main` - so when you're checking whether this new functionality is present, the check needs to be something like `indexVersion.between(BACKPORT_VERSION, 9_000_00_0) || indexVersion >= MAIN_CONSTANT`